app/src/main/java/io/gnosis/safe/notifications/models/PushNotification.kt

```diff
@@ -0,0 +1,94 @@
+package io.gnosis.safe.notifications.models
+
+import pm.gnosis.model.Solidity
+import pm.gnosis.utils.asEthereumAddress
+import java.math.BigInteger
+
+sealed class PushNotification(
+    val type: String
+) {
+
+    abstract val safe: Solidity.Address
+
+    data class NewConfirmation(
+        override val safe: Solidity.Address,
+        val safeTxHash: String,
+        val owner: Solidity.Address
+    ) : PushNotification(TYPE) {
+        companion object {
+            const val TYPE = "NEW_CONFIRMATION"
+            fun fromMap(params: Map<String, String>) =
+                NewConfirmation(
+                    params.getOrThrow("address").asEthereumAddress()!!,
+                    params.getOrThrow("safeTxHash"),
+                    params.getOrThrow("owner").asEthereumAddress()!!
+                )
+        }
+    }
+
+    data class ExecutedTransaction(
+        override val safe: Solidity.Address,
+        val safeTxHash: String,
+        val failed: Boolean
+    ) : PushNotification(TYPE) {
+        companion object {
+            const val TYPE = "EXECUTED_MULTISIG_TRANSACTION"
+            fun fromMap(params: Map<String, String>) =
+                ExecutedTransaction(
+                    params.getOrThrow("address").asEthereumAddress()!!,
+                    params.getOrThrow("safeTxHash"),
+                    params.getOrThrow("failed").toBoolean()
+                )
+        }
+    }
+
+    data class IncomingToken(
+        override val safe: Solidity.Address,
+        val txHash: String,
+        val value: BigInteger,
+        val tokenAddress: Solidity.Address,
+        val tokenId: String? // for ERC721 tokens
+    ) : PushNotification(TYPE) {
+        companion object {
+            const val TYPE = "INCOMING_TOKEN"
+            fun fromMap(params: Map<String, String>) =
+                IncomingToken(
+                    params.getOrThrow("address").asEthereumAddress()!!,
+                    params.getOrThrow("txHash"),
+                    params.getOrThrow("value").toBigInteger(),
+                    params.getOrThrow("tokenAddress").asEthereumAddress()!!,
+                    params["tokenId"]
```

Comment:
this one is the only not `getOrThrow`

Reply:
we get INCOMMING_TOKEN type for both ERC20 and ERC721 notifications. `tokenId` is null for ERC20. Currently, this is the only way to distinguish between them